6-AdvancedScenarios/4-hybrid-SPA/README.md

```diff
@@ -0,0 +1,409 @@
+1. All APIs have to publish a minimum of one [scope](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code), also called [Delegated Permissions](https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#permission-types), for the client's to obtain an access token successfully. To publish a scope, follow these steps:
+   * Select **Add a scope** button open the **Add a scope** screen and Enter the values as indicated below:
+        * For **Scope name**, use `access_as_user`.
+        * Select **Admins and users** options for **Who can consent?**.
+        * For **Admin consent display name** type `Allow the app msal-hybrid-spa to [ex, read ToDo list items] as the signed-in user`.
+        * For **Admin consent description** type `Allow the application to [ex, read ToDo list items] as the signed-in user.`
+        * For **User consent display name** type `[ex, Read ToDo list items] as yourself`.
+        * For **User consent description** type `Allow the application to [ex, Read ToDo list items] as the signed-in user on your behalf.`
+        * Keep **State** as **Enabled**.
+        * Select the **Add scope** button on the bottom to save this scope.
+1. Select the `Manifest` blade on the left.
+   * Set `accessTokenAcceptedVersion` property to **2**.
+   * Set the `optionalClaims` property as shown below to request client capabilities claim `idToken`:
+   ```json
+   "optionalClaims": {
+        "idToken": [
+            {
+                "name": "sid",
+                "source": null,
+                "essential": false,
+                "additionalProperties": []
+            },
+            {
+                "name": "login_hint",
+                "source": null,
+                "essential": false,
+                "additionalProperties": []
+            }
+        ],
+        "accessToken": [],
+        "saml2Token": []
+    }
+   ```
+
+   * Click on **Save**.
+
+#### Configure the service app (msal-hybrid-spa) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `.env` file.
+1. Find the key `Enter_the_Application_Id_Here` and replace the existing value with the application ID (clientId) of `msal-hybrid-spa` app copied from the Azure portal.
+1. Find the key `Enter_the_Tenant_Info_Here` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `Enter_the_Client_Secret_Here` and replace the existing value with the key you saved during the creation of `msal-hybrid-spa` copied from the Azure portal.
+1. Find the key `Redirect_URI` and replace the existing value with the Redirect URI for `msal-hybrid-spa`. (by default `http://localhost:5000/redirect`).
+
+#### Configure the client app (msal-hybrid-spa) to use your app registration
```

Comment:
Since both the backend and the frontend app uses the same app registration/credentials, can we somehow take in the config parameters from a single location? We shouldn`t ask the reader to edit the config files twice if not needed.